subsys/ipc/ipc_service/backends/ipc_icbmsg.c

```diff
@@ -541,7 +541,7 @@ static int send_release(struct backend_data *dev_data, const uint8_t *buffer,
  * @param[in] size		Actual size of the data, can be smaller than allocated,
  *				but it cannot change number of required blocks.
  *
- * @return			O or negative error code.
+ * @return			number of sent blocks or negative error code.
```

Comment:
I would stick to "non-negative value in case of success or negative error code".
In fact this function returns the result of underlying `send_control_message` which returns the result of underlying `icmsg_send` which returns as you fixed in `icmsg.h` file "Number of sent bytes." where the sent bytes refer rather to the size of control message.

Reply:
The ipc_service backends must follow the ipc_service API. ipc_service currently requires to return number of sent bytes. If we want to change it, we would need to change ipc_service API. Not only the ipc_icbmsg backend.

Reply:
This is now done within the `send` function, after Piotr's fix. This is the core of the fix. So `send` now follows the API.

`send_block` is internal to the ipc_icbmsg implementation. It can return anything and it would really nice if description of what this function returns met the code.

Reply:
Right, my mistake.